upnext/views.py

```diff
@@ -50,18 +51,35 @@ def party_detail(request, party_url):
     party_obj = Party.objects.get(url=party_url)
     party_tracks = party_obj.track_set.all()
     tracks_ordered = party_tracks.order_by('-score')
+    party_embed_url = 'https://embed.spotify.com/?uri=' + party_obj.uri
 
-    context = {'party': party_obj, 'tracks': tracks_ordered}
+    try:
+        print "i got here"
+        current = party_obj.track_set.get(current=True)
+        print current
+    except:
+        current = None
+
+    if request.user.username == party_obj.host and party_tracks:
+        current = scrape.get_currently_playing(party_obj, party_embed_url)
+        # print current, "current"
 
     if 'track_query' in request.GET:
         return track_search_results(request, request.GET['track_query'], party_obj)
 
     elif request.method == "POST":
         if 'track_up' in request.POST:
             tracks.upvote_track(request.POST['track_up'], party_obj)
+
         elif 'track_down' in request.POST:
             tracks.downvote_track(request.POST['track_down'], party_obj)
 
+        elif 'new_host' in request.POST:
+            party_obj.host = request.user.username
+            party_obj.host_display = request.POST['display_name']
+            party_obj.save()
+    print current, "current before return"
```

Comment:
Do we want to keep the print statements in this file? Not sure if they're super useful for debugging tbh